compact_str/src/repr/inline.rs

```diff
@@ -29,6 +32,25 @@ impl InlineString {
         InlineString { buffer }
     }
 
+    #[inline]
+    pub fn from_fmt(val: impl fmt::Display, len: usize) -> Self {
+        debug_assert!(len <= MAX_SIZE);
+
+        let mut buffer = [0u8; MAX_SIZE];
+
+        // set the length
+        buffer[MAX_SIZE - 1] = len as u8 | LENGTH_MASK;
+
+        // format the val into buffer.
+        //
+        // note: in the case where len == MAX_SIZE, we'll overwrite the len, but that's okay because
+        // when reading the length we can detect that the last byte is part of UTF-8 and return a
+        // length of MAX_SIZE
+        write!(&mut buffer[..], "{}", val).expect("fmt::Display incorrectly implemented!");
+
+        InlineString { buffer }
+    }
```

Comment:
This introduces another part of the code base that directly writes into the buffer and needs to know about the underlying bit layout, I'd like to prevent that. 

See my comment on `impl<T: fmt::Display> ToCompactStr for T` on how we can eliminate this

Reply:
Done!